airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/AbstractPostgresSourceDatatypeTest.java

```diff
@@ -293,11 +293,36 @@ protected void initTests() {
                 "08:00:2b:01:02:03:04:07")
             .build());
 
+    // Edge case of a column type being a NUMERIC data type
+    // with no decimal precision
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("numeric")
+            .fullSourceDataType("NUMERIC(38,0)")
+            .airbyteType(JsonSchemaType.INTEGER)
+            .addInsertValues("'33'")
+            .addExpectedValues("33")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("numeric")
+            .fullSourceDataType("NUMERIC(28,2)")
+            .airbyteType(JsonSchemaType.NUMBER)
+            .addInsertValues(
+                "'123'", "null", "'14525.22'")
+            // Postgres source does not support these special values yet
+            // https://github.com/airbytehq/airbyte/issues/8902
+            // "'infinity'", "'-infinity'", "'nan'"
+            .addExpectedValues("123", null, "14525.22")
+            .build());
+
     // Blocked by https://github.com/airbytehq/airbyte/issues/8902
     for (final String type : Set.of("numeric", "decimal")) {
       addDataTypeTestData(
           TestDataHolder.builder()
               .sourceType(type)
+              .fullSourceDataType("NUMERIC(20,7)")
               .airbyteType(JsonSchemaType.NUMBER)
               .addInsertValues(
                   "'123'", "null", "'1234567890.1234567'")
```

Comment:
@prateekmukhedkar, for this test case in its current form, the `numeric` source type will have a 0 decimal scale since the fullSourceDataType wasn't specified. With my change, `1234567890.1234567` will be truncated to `1234567890` I've updated this test case to `NUMERIC(20,7)` to accommodate the chosen value.